curated_transformers/pipe.py

```diff
@@ -258,13 +262,15 @@ def predict(self, docs: Iterable[Doc]):
             ]
         return self.model.predict(docs)
 
-    def set_annotations(self, docs: Sequence[Doc], tokvecses) -> None:
+    def set_annotations(
+        self, docs: Sequence[Doc], trf_output: TransformerModelOutput
+    ) -> None:
         """Modify a batch of documents, using pre-computed scores.
         docs (Iterable[Doc]): The documents to modify.
         tokvecses: The tensors to set, produced by Tok2Vec.predict.
         DOCS: https://spacy.io/api/tok2vec#set_annotations
         """
-        for doc, tokvecs in zip(docs, tokvecses):
+        for doc, tokvecs in zip(docs, trf_output.all_outputs):
```

Comment:
I wonder if we can do nicer here, by wrapping the per-doc outputs in another class. E.g. `DocTransformerModelOutput` (there is probably a better name). Then we don't need the `-1` indexing in other places, but could just have a property to access the last hidden layer.

Reply:
Sounds good, I'll add that.